client.js

```diff
@@ -13,9 +13,15 @@ var options = {
   overlayStyles: {},
   overlayWarnings: false,
   ansiColors: {},
+  beforeUpdateApplyThenable() {
+    return {
+      then(cb) {
+        cb()
+      }
+    }
+  } 
 };
 if (__resourceQuery) {
-  var { URLSearchParams } = require('url');
```

Comment:
we hit node builtin polyfill not set with webpack 5, but `URLSearchParams` seems to be available in all browsers we support natively.

I also realized there is https://github.com/gatsbyjs/webpack-hot-middleware/pull/2 PR that seems to try to do something about this require as well, so might be good to resolve those 2 together